clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -234,7 +234,24 @@
 
     /** <code>partitioner.class</code> */
     public static final String PARTITIONER_CLASS_CONFIG = "partitioner.class";
-    private static final String PARTITIONER_CLASS_DOC = "Partitioner class that implements the <code>org.apache.kafka.clients.producer.Partitioner</code> interface.";
+    private static final String PARTITIONER_CLASS_DOC = "A class to use to determine which partition to be send to when produce the records. Available options are:" +
+        "<ul>" +
+            "<li><code>org.apache.kafka.clients.producer.internals.DefaultPartitioner</code>: The default partitioner. Works with the strategy:" +
+                "<ul>" +
+                    "<li>If a partition is specified in the record, use it</li>" +
+                    "<li>If no partition is specified but a key is present choose a partition based on a hash of the key</li>" +
+                    "<li>If no partition or key is present choose the sticky partition that changes when the batch is full.</li>" +
+                "</ul>" +
+            "</li>" +
+            "<li><code>org.apache.kafka.clients.producer.RoundRobinPartitioner</code>: This partitioning strategy can be used when user wants to distribute the writes to all partitions equally.</li>" +
```

Comment:
Would it be possible to describe the strategy in more detail here? Something along the lines of how each record is assigned a new partition?

I also wonder if it suffices to describe the RoundRobinPartitioner  as the `strategy can be used when user wants to distribute the writes to all partitions equally`. In many cases, the DefaultPartitioner also tries to do this, though I am aware of the cases where it doesn't. (https://issues.apache.org/jira/browse/KAFKA-10888).